dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/taglib/GlobalTagLibraryCache.java

```diff
@@ -582,23 +579,45 @@ private void loadTldFromClassloader(GlobalTagLibConfig globalTagLibConfig, TldPa
                         }
-                        tli.setURI(tli.getReliableURN());
-                        put(tli.getReliableURN(), tli);
-                        tldPathConfig.setUri(tli.getReliableURN());
+                        /* If customURIUsed is true, we ignore whatever the URI is in the TLD 
+                         * and, instead, we use the URI specified via the TldPathConfig constructor
+                         * See OLGH 26891
+                         */
+                        if(!tldPathConfig.isCustomURIUsed()){ 
+                            // Custom URI Is NOT Used -- Continue as Normal (As Before)
+                            if (containsKey(tli.getReliableURN()) == false) {
+                                tli.setURI(tli.getReliableURN());
+                                put(tli.getReliableURN(), tli);
+                                tldPathConfig.setUri(tli.getReliableURN());
+                            }
+                        } else { // Custom URI is used, so override the one set in the TagLibraryInfoImpl
+                            if (containsKey(tldPathConfig.getUri()) == false) {
+                                tli.setURI(tldPathConfig.getUri());
+                                tli.setReliableURN(tldPathConfig.getUri());
+                                put(tldPathConfig.getUri(), tli);
+                            }
+                        }
                         eventListenerList.addAll(tldParser.getEventListenerList());
-                    }
                 } else {
+                    Object location = tldInputSource.getAbsoluteURL();
+                    if(location == null){
+                        location =  tldPathConfig.getTldPath();
+                    }
                     if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.WARNING)) {
-                        logger.logp(Level.WARNING, CLASS_NAME, "loadTldFromClassloader", "jsp warning failed to find a uri tag in [" + tldInputSource.getAbsoluteURL() + "]");
+                        logger.logp(Level.WARNING, CLASS_NAME, "loadTldFromClassloader", "jsp warning failed to find a uri tag in [" + location + "] (loaded from " + globalTagLibConfig + ")");
                     }
                 }
             } catch (JspCoreException e) {
+                    Object location = tldInputSource.getAbsoluteURL();
+                    if(location == null){
+                        location = tldPathConfig.getTldPath();
+                    }
                 if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.WARNING)) {
-                    logger.logp(Level.WARNING, CLASS_NAME, "loadTldFromClassloader", "jsp warning failed to load tld in jar. uri = [" + tldInputSource.getAbsoluteURL() + "]", e);
+                    logger.logp(Level.WARNING, CLASS_NAME, "loadTldFromClassloader", "jsp warning failed to load tld in " + globalTagLibConfig + ". uri = [" + location + "]", e);
```

Comment:
What does the new warning looks like vs the old in the logs?

Reply:
I think I go a null here, so I changed it around. However, I undid the change since if the getTldPath is wrong, then the file can't be parsed and throws an earlier error.